core-bundle/src/Migration/Version411/TwoFactorBackupCodesMigration.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Migration\Version411;
+
+use Contao\CoreBundle\Migration\AbstractMigration;
+use Contao\CoreBundle\Migration\MigrationResult;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @internal
+ */
+class TwoFactorBackupCodesMigration extends AbstractMigration
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function shouldRun(): bool
+    {
+        $schemaManager = $this->connection->getSchemaManager();
+
+        if (!$schemaManager->tablesExist(['tl_user', 'tl_member'])) {
+            return false;
+        }
+
+        $users = $this->getAffectedRowsForTable('tl_user');
+        $members = $this->getAffectedRowsForTable('tl_member');
+
+        return \count($users) || \count($members);
+    }
+
+    public function run(): MigrationResult
+    {
+        $this->updateBackupCodes('tl_user');
+        $this->updateBackupCodes('tl_member');
+
+        return $this->createResult(true);
+    }
+
+    private function updateBackupCodes(string $table): void
+    {
+        $rows = $this->getAffectedRowsForTable($table);
+
+        if (!\count($rows)) {
+            return;
+        }
+
```

Comment:
This is not necessary I think because the foreach would just do nothing.

```suggestion
```

Reply:
Fixed in 5d072f6dd.